pythonwhat/Test.py

```diff
@@ -117,34 +123,64 @@ def is_equal(x, y):
     try:
         if areinstance(x, y, (str, int, float, bool, type(None))):
             return x == y
+
         if is_collection_of_primitives(x):
             return x == y
+
         if areinstance(x, y, (Exception,)):
             # Types of errors don't matter (this is debatable)
             return str(x) == str(y)
 
+        if areinstance(x, y, (list, tuple,)):
+            if len(x) != len(y):
+                return False
+            return all(is_equal(x_element, y_element) for x_element, y_element in zip(x, y))
+
+        if areinstance(x, y, (map, filter,)):
+            x_list, y_list = list(x), list(y)
+            if len(x_list) != len(y_list):
+                return False
+            return all(is_equal(x_element, y_element) for x_element, y_element in zip(x_list, y_list))
+
+        if areinstance(x, y, (set,)):
+            return is_equal(sorted(x), sorted(y))
+
+        if areinstance(x, y, (dict,)):
+            if x.keys() != y.keys():
+                return False
+            return all(is_equal(x[key], y[key]) for key in x)
+
         # Delay importing pandas / numpy until absolutely necessary. This is important for performance in Pyodide.
-        import pandas as pd
-        from pandas.testing import assert_frame_equal, assert_series_equal
-        import numpy as np
-
-        if areinstance(x, y, (np.ndarray, dict, list, tuple)):
-            np.testing.assert_equal(x, y)
-            return True
-        elif areinstance(x, y, (map, filter)):
```

Comment:
Are we still covering this conditional statement?

Reply:
Yes, L139